managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1411,9 +1411,21 @@ synchronized void ledgerClosed(final LedgerHandle lh) {
             log.debug("[{}] Ledger has been closed id={} entries={}", name, lh.getId(), entriesInLedger);
         }
         if (entriesInLedger > 0) {
-            LedgerInfo info = LedgerInfo.newBuilder().setLedgerId(lh.getId()).setEntries(entriesInLedger)
-                    .setSize(lh.getLength()).setTimestamp(clock.millis()).build();
-            ledgers.put(lh.getId(), info);
+            synchronized (ledgers) {
```

Comment:
We are not using synchronized on ledgers variable in other points in this class.
It looks to me as an inconsistent synchronisation.

Reply:
Thanks, eolivelli, I think you are right.

I checked the code and found out all situations that change the `ledgers` in the `ManagedLedgerImpl`, for a ledger its `LedgerInfo` update operations don't exist competition.

    1. When the ManagedLedgerImpl initialized, recover metadata from the metadata store.
    2. When creating a new ledger.
    3. When the ledger closed.
    4. When the offloaded ledger data should be deleted from bk.
    5. When preparing for offload.
    6. When offloading complete.

I think the `LedgerInfo` should be initialized when the ledger created, after this, the `LedgerInfo` should be updated not be created.

PTAL